packages/ui/src/components/va-select/VaSelect.vue

```diff
@@ -294,20 +294,20 @@ export default defineComponent({
 
     const valueComputed = computed<SelectOption | SelectOption[]>({
       get () {
-        const value = getOptionByValue(props.modelValue)
-
         if (props.multiple) {
-          if (!value) {
+          if (!props.modelValue) {
             return []
           }
 
-          if (!Array.isArray(value)) {
-            return [value]
+          if (!Array.isArray(props.modelValue)) {
+            return [getOptionByValue(props.modelValue)]
           }
 
-          return value.map(getValue)
+          return props.modelValue.map((v) => getOptionByValue(v))
```

Comment:
We can pass the `getOptionByValue` method directly into the map's callback. It doesn't accept any other arguments rather than value.
```suggestion
          return props.modelValue.map(getOptionByValue)
```